### PR TITLE
Refactor parser

### PR DESCRIPTION
**Pull request by 0918nobita**
Refactor `parseElement` and related functions

### CODE REVIEW

File: psvg.ts [hunk 1]
```diff
@@ -440,12 +449,4 @@ export function compilePSVG(psvg:string):any{
...
 }
 
-if (typeof window !== 'undefined') {
-  window.addEventListener('load',function(){
-    const psvgs = document.getElementsByTagName("PSVG");
-    for (let i = 0; i < psvgs.length; i++){
-      psvgs[i].outerHTML = compilePSVG(psvgs[i].outerHTML);
-    }
-  })
-}
```
Comment (antfu) [inline psvg.ts:450]: Wondering why is this removed?

Comment (0918nobita) [reply psvg.ts:450]: I did not edit that file directly; when I used the `npm run build` command, it was overwritten on its own.

Comment (0918nobita) [reply psvg.ts:450]: Due to my operational error it seems to have removed.

Comment (0918nobita) [reply psvg.ts:450]: I've fixed it.

**Comment by LingDong-** [comment]
LGTM. 
Neat, thanks for the refactoring!

**Comment by LingDong-** [comment]
uhm... looks like this is breaking the examples on https://psvg.netlify.app/

image

**Comment by LingDong-** [comment]
OK fixed. ef5dcaeb1755aa59bb8a44d922f53c99c70ff386

This line:

```js
33          const matchAll = attrsStr.matchAll || ((re: RegExp) => {
```

needs to be changed to:

```js
33          const matchAll = (re:RegExp)=>attrsStr.matchAll(re) || ((re: RegExp) => {
```

A really subtle JS quirk :)

**Comment by 0918nobita** [comment]
Thank you for the quick response!

**Comment by LingDong-** [comment]
No problem! Actually I made a mistake too, it should be

```js
33          const matchAll = attrsStr.matchAll ? (re:RegExp)=>attrsStr.matchAll(re) : ((re: RegExp) => {
```

Fixed in 4e79e34078624f1ea5ed2f85250df25d9aacf5b9

😂
